### PR TITLE
Refactor: unify indicators behaviour

### PR DESCRIPTION
**Pull request by tirz**

### CODE REVIEW

File: src/indicators/maximum.rs [hunk 1]
```diff
@@ -28,34 +28,30 @@ use serde::{Deserialize, Serialize};
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(Debug, Clone)]
 pub struct Maximum {
-    n: usize,
-    vec: Vec<f64>,
+    period: usize,
     max_index: usize,
     cur_index: usize,
+    deque: Box<[f64]>,
```
Comment (greyblake) [inline src/indicators/maximum.rs:34]: Could you please explain this change `Vec<f64>` VS `Box<[f64]>` ?

**Comment by greyblake** [comment]
@tirz  Thank you, looks pretty good! I have thoughts about replacing `u32` with `usize`, but was not sure. In your changes I see, that avoiding the casting u32 -> usize justifies this.
Could pls explain `Box<[f64]>` thing to me and I'm ready to merge.

**Comment by tirz** [comment]
Hi @greyblake, sorry for the delay.

The change from a vector to an owned array is because we do not need the dynamically sized feature provided by the vector (nether for the `add()` and `remove()` functions) since an indicator set his period while he is initialized.

We can argue that an array is lighter and faster to read/write than a vector but in the case of the crate ta-rs, we should not see any benefice on the benchmark...
The change is here only to show clearly that the length of the period is fixed.

Please, note that I also refactorized `EfficiencyRatio` and `RateOfChange` to replace the use of the `VecDeque` by a `Box<[f64]>` on my local copy of ta-rs.
If you like this change, I will be more than happy to fill a PR.

**Comment by greyblake** [comment]
@tirz Thanks for an explanation. So basically `Box<[f64]>` is just an array allocated on heap, as I understood it.
You can open PRs for `EfficiencyRatio` and `RateOfChange`, I appreciate you work.
Thanks!

**Pull request merged**